crates/solidity/testing/sanctuary/src/reporting.rs

```diff
@@ -44,7 +44,8 @@ impl Reporter {
 
         let failed_tests = self.failed_tests.load(Ordering::Relaxed);
         let total_tests = self.total_tests.load(Ordering::Relaxed);
-        let failure_percent = (100_f64 * (failed_tests as f64) / (total_tests as f64)) as usize;
+        #[allow(clippy::cast_possible_truncation, clippy::cast_precision_loss)]
+        let failure_percent = (100_f64 * (failed_tests as f64) / (total_tests as f64)) as isize;
```

Comment:
I might be missing something. Why does that need the signed `isize`?

Reply:
I think it is virtually impossible for this to actually happen, but I'm happy to turn the rule on if you think it is helpful otherwise. But if we are enabling this rule, would it make sense to use the suggested `try_from()` instead of `#[allow]`ing it?

Reply:
I have to admit, I wanted to quickly appease the `cast_sign_loss` here and not break the allows into multi-line. The code is fairly reasonable and the edge cases are, as you said, virtually impossible. Our code uses floats... precisely in this one single instance, so could silence the lints as well. What do you think?

Reply:
Sounds good. I'm happy to give it a try, and we can always disable later if it is too noisy.